### PR TITLE
Add clusters with replicas from all replica groups

### LINKED ISSUE #62725: Allow clusterAll over all replicas across different ReplicaGroup as an option
> (you don't have to strictly follow this form)

**Use case**

With replica group, all clusterAllReplicas queries will be run within the current replicaGroup. For monitoring for the entire cluster, we may need to access all replicas in the cluster for some clusterAllReplicas system table queries. Thus, it would be nice to support an option for clusterAllReplicas to reach every replicas across replicaGroup.

**Describe the solution you'd like**

`SELECT * from clusterAllReplicas('default', system.replicas)` should give within replicaGroup
`SELECT * from clusterAllReplicas('all', system.replicas)` could give results across replicaGroups

**Describe alternatives you've considered**
NA

**Additional context**
https://github.com/ClickHouse/ClickHouse/issues/53620

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
If "replica group" is configured for a `Replicated` database, automatically create a cluster that includes replicas from all groups.

Closes https://github.com/ClickHouse/ClickHouse/issues/62725 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>CI Settings</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Run these jobs only (required builds will be added automatically):
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with aarch64
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Deny these jobs:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

</details>

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit ed8434c80fb4f8f24cfe787715b025fc23ca6c01 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64312/ed8434c80fb4f8f24cfe787715b025fc23ca6c01/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -173,13 +173,40 @@ ClusterPtr DatabaseReplicated::tryGetCluster() const
     return cluster;
 }
 
-void DatabaseReplicated::setCluster(ClusterPtr && new_cluster)
+ClusterPtr DatabaseReplicated::tryGetAllGroupsCluster() const
 {
     std::lock_guard lock{mutex};
-    cluster = std::move(new_cluster);
+    if (replica_group_name.empty())
+        return nullptr;
```
Comment (sichenzhao) [inline src/Databases/DatabaseReplicated.cpp:180]: should it just return regular getClusterImpl when we accidentally call tryGetAllGroups on a clickhouse instance without replicaGroup?

**Review by serxa — APPROVED** [review]
Maybe add a small test?

**Review by sichenzhao — APPROVED** [review]

**Comment by tavplubix** [comment]
AST fuzzer (debug) - https://github.com/ClickHouse/ClickHouse/issues/56640
Integration tests (aarch64) [5/6] - https://github.com/ClickHouse/ClickHouse/issues/64451
Stress test (debug) - https://github.com/ClickHouse/ClickHouse/issues/64936
Stress test (msan) - https://github.com/ClickHouse/ClickHouse/issues/64932

**Comment by sichenzhao** [comment]
Found a potential problem here in dev after change is merged - `RENAME/DROP/UPDATE DATABASE` query doesn't work on all clusters:

in DDL queue, we automatically append `ON CLUSTER DEFAULT` for those DDL queries, but after this change and replica_group config change in hydra, it's limited to replica_groups now. 

I am wondering if we could do `ON CLUSTER all_replicas.default` on those DDL queries and apply to all groups... I manually applied those queries but get errors on .`

**Comment by sichenzhao** [comment]
fixed this too https://github.com/ClickHouse/data-plane-application/issues/9804

**Pull request merged**